crates/core/src/operations/optimize.rs

```diff
@@ -739,18 +740,18 @@ impl MergePlan {
                     app_metadata.insert("operationMetrics".to_owned(), map);
                 }
 
-                table.update().await?;
```

Comment:
Instead of removing the update completely and forcing users to update their table we can move this update to right before the operation returns, right?